grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -725,14 +727,73 @@ class QueryService implements Status {
+            result = findIssuesSearchItemsDTO(ids)
         }
         result
     }
 
+    /**
+     * Find issues and create DTOs to narrow the amount of data used in the search process
+     * @param issueIds
+     * @return
+     */
+    Set<IssueSearchItemDTO> findIssuesSearchItemsDTO(ArrayList<Integer> issueIds) {
+        final String query = "SELECT i.id id, " +
+                "i.project_key projectKey, " +
+                "i.type type, " +
+                "i.status status, " +
+                "i.summary summary, " +
+                "i.reporter reporter, " +
+                "i.update_date updated, " +
+                "i.expiration_date expirationDate, " +
+                "iep.* " +
+                "FROM issue i LEFT JOIN issue_extra_property iep " +
+                "ON (iep.project_key = i.project_key AND iep.name in ('pm','pi','collaborator')) " +
+                "WHERE i.id IN (" + issueIds.join(",") + ")"
+
+        Set<IssueSearchItemDTO> resultDTO = new HashSet<IssueSearchItemDTO>()
+        IssueSearchItemDTO issueSearchItemDTO
+        String currentProjectKey = ""
+
+        getSqlConnection().rows(query).each {
+            if (it.get("projectKey") == currentProjectKey) {
+                if (it.get("type") != IssueType.CONSENT_GROUP.name) {
+                    def extraProp = issueSearchItemDTO.extraProperties.get(it.get("name").toString()) ?
+                            issueSearchItemDTO.extraProperties.get(it.get("name").toString()) : []
+                    extraProp.push(it.get("value").toString())
+                }
+            } else {
+                if (currentProjectKey != "") {
+                    resultDTO.add(issueSearchItemDTO)
+                }
+                currentProjectKey = it.get("projectKey")
+                issueSearchItemDTO = new IssueSearchItemDTO()
+
+                issueSearchItemDTO.setId((Integer)it.get("id"))
+                issueSearchItemDTO.setProjectKey(it.get("projectKey").toString())
+                issueSearchItemDTO.setType(it.get("type").toString())
+                issueSearchItemDTO.setStatus(it.get("status").toString())
+                issueSearchItemDTO.setSummary(it.get("summary").toString())
+                issueSearchItemDTO.setReporter(it.get("reporter").toString())
+                if (it.get("expirationDate") != null) {
+                    issueSearchItemDTO.setExpirationDate(it.get("expirationDate"))
+                }
+                if (it.get("updated") != null) {
+                    issueSearchItemDTO.setUpdateDate(it.get("updated"))
+                }
+                if (it.get("type") != IssueType.CONSENT_GROUP.name) {
+                    def extraProp = issueSearchItemDTO.extraProperties.size() != 0 ? issueSearchItemDTO.extraProperties.get(it.get("name").toString()) : []
+                    extraProp.push(it.get("value").toString())
+                    issueSearchItemDTO.extraProperties.put(it.get("name").toString(), extraProp)
```

Comment:
I'm finding this logic confusing ... can you comment on what's happening here?

Reply:
Any feedback on this question? This looks confusing enough to warrant some kind of comment that describes what it's doing.